doc: add note in BUILDING.md about running `make distclean` (#31542)
nodejs/node: BUILDING.md

The above requires that `python` resolves to a supported version of
Python. See [Prerequisites](#prerequisites).

> If you run into a `file not found` error, please try

Reviewer: I'm not sure it's worth being this specific unless this is the only error seen with a stale build.

Author: Modified the messaging accordingly.

> A stale build can lead to issues, for example a `file not found` error.

Reviewer: ```suggestion
A stale build can lead to issues, for example a "file not found" error.
```

Reviewer 2: Building on the previous suggestion:

```suggestion
A stale build can lead to issues such as "file not found" errors.
```

> To fix these, please try

Reviewer 2: Move this to the end of the previous paragraph rather than having it be a new paragraph referring to "these" from the previous paragraph.
